// @workspace/api-zod/src/generated/types/surveySubmission.ts
/**
 * Generated by orval v8.5.3 🍺
 * Do not edit manually.
 * Api
 * College Student Lifestyle & Productivity Survey API
 * OpenAPI spec version: 0.1.0
 */
import type { SurveySubmissionClassAttendance } from "./surveySubmissionClassAttendance";
import type { SurveySubmissionDistractionsItem } from "./surveySubmissionDistractionsItem";
import type { SurveySubmissionFeelsProductive } from "./surveySubmissionFeelsProductive";
import type { SurveySubmissionProductivityHabitsItem } from "./surveySubmissionProductivityHabitsItem";
import type { SurveySubmissionSleepHours } from "./surveySubmissionSleepHours";
import type { SurveySubmissionStudyHoursPerWeek } from "./surveySubmissionStudyHoursPerWeek";

export interface SurveySubmission {
  /** Student's major */
  major: string;
  /** How often they attend class */
  classAttendance: SurveySubmissionClassAttendance;
  /** Weekly study hours range */
  studyHoursPerWeek: SurveySubmissionStudyHoursPerWeek;
  /** Main distractions while studying */
  distractions: SurveySubmissionDistractionsItem[];
  /** Whether they feel productive most days */
  feelsProductive: SurveySubmissionFeelsProductive;
  /** Hours of sleep per night */
  sleepHours: SurveySubmissionSleepHours;
  /** Things they do to stay productive */
  productivityHabits: SurveySubmissionProductivityHabitsItem[];
  /** Optional text about what they'd change about their routine */
  routineChange?: string | null;
}
